### DISCUSSION TITLE
How to display all tables as inline?

### DISCUSSION THREAD

**Discussion by zmitchell** [toml-rs/toml#644] [Q&A]
Let's say I have a table called `foo` and under that are other tables:
```
[foo]
[foo.entry1]
[foo.entry2]
etc
```
If I parse this and want to dump it back out so that all of the sub-tables are displayed as inline tables (shown below), how would I do that?
```
[foo]
entry1 = {}
entry2 = {}
etc
```

**Comment by epage** [comment]
Currently, `toml` doesn't have much formatting control.  Rather than bake-in control for each one off case (and becoming complex), I would instead encourage deserializing to a `toml_edit::Document` and using `toml_edit::visit_mut` to style the output as desired.

**Comment by zmitchell** [comment]
Ah, I think the missing piece here was that I didn't know the `InlineTable` type is represented with `Item::Value(Value::InlineTable(_))` so I couldn't figure out how to represent an `InlineTable` with an `Item`. I figured this out looking at https://github.com/toml-rs/toml/blob/main/crates/toml_edit/examples/visit.rs.

P.S. - the link to `examples/visit.rs` on [this page](https://docs.rs/toml_edit/latest/toml_edit/visit_mut/index.html) is broken.

**Reply by epage** [reply]
Links should be fixed in 6c8885e
